mlir/lib/Conversion/AffineToStandard/AffineToStandard.cpp

```diff
@@ -349,14 +361,62 @@ class AffineForLowering : public OpRewritePattern<AffineForOp> {
     Value lowerBound = lowerAffineLowerBound(op, rewriter);
     Value upperBound = lowerAffineUpperBound(op, rewriter);
     Value step = rewriter.create<ConstantIndexOp>(loc, op.getStep());
-    auto f = rewriter.create<scf::ForOp>(loc, lowerBound, upperBound, step);
+    SmallVector<Value, 4> iterArgs(op.getIterOperands().begin(),
+                                   op.getIterOperands().end());
+    scf::ForOp f;
+    if (iterArgs.size() > 0) {
+      f = rewriter.create<scf::ForOp>(loc, lowerBound, upperBound, step,
+                                      iterArgs);
```

Comment:
I think you can directly pass `op.getIterOperands()` here without materializing it into a SmallVector.

Reply:
Done.